### ISSUE TITLE
Sync-free workflow?

### ISSUE THREAD

**Issue by cpitclaudel** [orgzly/orgzly-android#668]
- [x] I have searched for existing issues that may be the same as or related to mine.

Thanks for the hard work on Orgzly.  I use Orgzly to edit notes on my phone, and Emacs to edit notes on my computer.  I synchronize changes between both using syncthing.  

I keep running into issues, however: sometimes Orgzly's auto-sync recreates notes that I deleted on my computer, sometimes I forget to sync manually and I overwrite changes made on my computer, sometimes I get multiple copies of the same file with `(1)` or `(2)` appended, etc..

**Feature request:** I'd like Orgzly to *not* maintain a separate copy of my notes in the app; instead, I'd like it to act like other text editors or photo galleries on my phone: when I start Orgzly it would show all files in `Documents/Orgzly`, and every time I edit a notebook it would overwrite the corresponding file.  When I open a notebook it would just read the file from disk.  (I realize that I will run into issues if I edit a note at the same time in two places, but I'm ready to live with that, and that problem exists with all other apps.) This way, with Orgzly reading straight from my SD card and writing there, there would be no need for two levels of synchronization (Orgzly ­→ SD card followed by SD card → PC). 

In other words, I'd like to use Orgzly as an org file editor, without using the synchronization, notebook management, or even task displays.  Can this be done?

Thanks again :)

**Comment by nevenz** [comment]
Since data is stored in the database (for faster editing and searching), there's no way to avoid the step of syncing data with the local file.

Do you have auto-sync enabled? There's a lot of room for improvement there. Anything specific that doesn't work for you?

There's an idea to try and keep notebook file updated after every change, without having to export all notebook's notes every time.

**Comment by cpitclaudel** [comment]
> Since data is stored in the database (for faster editing and searching), there's no way to avoid the step of syncing data with the local file.

Would it be very costly to check if the data is out of date when a note is opened, and to sync the note as soon as it is closed?

> There's an idea to try and keep notebook file updated after every change, without having to export all notebook's notes every time.

That sounds nice.

> Do you have auto-sync enabled? There's a lot of room for improvement there. Anything specific that doesn't work for you?

I do.  I think I've mostly had issues with renames and deletions.  I rename or delete notes on my computer, and they pop back up after I sync with Orgzly.  Maybe I'm doing something wrong?

**Comment by nevenz** [comment]
> Would it be very costly to check if the data is out of date when a note is opened, and to sync the note as soon as it is closed?

Probably, but it's planned as an option. Check https://github.com/orgzly/orgzly-android/issues/8#issuecomment-546687187 for the full list.

> I think I've mostly had issues with renames and deletions. I rename or delete notes on my computer, and they pop back up after I sync with Orgzly.

Notes or notebooks?  Support for deleting notebooks (when remote notebook has been removed) is yet to be implemented (#287).

**Comment by cpitclaudel** [comment]
> Probably, but it's planned as an option. Check #8 (comment) for the full list.

Neat, thanks a lot!

> Notes or notebooks? Support for deleting notebooks (when remote notebook has been removed) is yet to be implemented (#287).

Notebooks, indeed.

**Comment by nevenz** [comment]
I'll close this as a duplicate (of #287 and #434). You can open a new issue for a specific feature if there isn't one already. Thanks!
